src/nnet/nnet-randomizer.cc

```diff
@@ -64,9 +64,11 @@ void MatrixRandomizer::AddData(const CuMatrixBase<BaseFloat>& m) {
   }
   // extend the buffer if necessary,
   if (data_.NumRows() < data_end_ + m.NumRows()) {
-    CuMatrix<BaseFloat> data_aux(data_);
-    // Add extra 1000 rows, so we don't reallocate soon:
-    data_.Resize(data_end_ + m.NumRows() + 1000, data_.NumCols());
+    // CuMatrix -> Matrix -> CuMatrix (needs less GPU memory),
+    Matrix<BaseFloat> data_aux(data_);
+    // Add extra 3% rows, so we don't reallocate soon:
```

Comment:
typically in situations like this (for a growing buffer) one would increase by a factor closer to 1.5 or 2, like stl vector does.  Or simply allocate the rows separately as vectors, if feasible- but there may be some reason why this is impractical.  I'm not sure if something is different here...

Reply:
The 'data_' buffer is supposed to have a 'fixed size', plus extra frames coming from 'last utterance', plus the 3% as a reserve for even longer utterances. The buffer is quite large (32k rows or more) and the dimensions can be large as well. So the 3% are there to give some extra space without consuming too much memory...